### PR TITLE
[Filter] add common functions

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Add common functions for tensor-filter and filter-single.
1. add struct for common property
2. reduce duplicated code of tensor-filter

Signed-off-by: Jaeyun Jung <jy1210.jung@samsung.com>

### CODE REVIEW

File: api/capi/include/tensor_filter_single.h [hunk 1]
```diff
@@ -54,15 +55,7 @@ struct _GTensorFilterSingle
...
-  void *privateData; /**< NNFW plugin's private data is stored here */
-  GstTensorFilterProperties prop; /**< NNFW plugin's properties */
-  const GstTensorFilterFramework *fw; /**< The implementation core of the NNFW. NULL if not configured */
-
-  /* internal properties for tensor_filter_single */
-  gboolean silent; /**< Verbose mode if FALSE. int instead of gboolean for non-glib custom plugins */
-  gboolean started; /**< filter has been started */
-  GstTensorsConfig in_config; /**< input tensor info */
-  GstTensorsConfig out_config; /**< output tensor info */
+  GstTensorFilterPrivate priv; /**< Internal properties for tensor-filter */
```
Comment (myungjoo) [inline api/capi/include/tensor_filter_single.h]: Good!

File: api/capi/include/tensor_filter_single.h [hunk 2]
```diff
@@ -30,6 +30,7 @@
 
 #include <nnstreamer/nnstreamer_subplugin.h>
 #include <nnstreamer/nnstreamer_plugin_api_filter.h>
+#include <nnstreamer/tensor_filter/tensor_filter_common.h>
```
Comment (myungjoo) [inline api/capi/include/tensor_filter_single.h]: Does this really need to include tensor_filter_common.h?
It'd be better if we can include tensor_filter_common.h in *.c only.

File: api/capi/include/tensor_filter_single.h (hunk 1 above)
Comment (myungjoo) [reply api/capi/include/tensor_filter_single.h]: However, if the intention is to hide detailed from app writers, exposing tensor_filter_common.h is not good.

**Comment by jaeyun-jung** [comment]
V2 (2019-09-17)
- remove unnecessary declaration in common header

File: api/capi/include/tensor_filter_single.h (hunk 2 above)
Comment (myungjoo) [reply api/capi/include/tensor_filter_single.h]: ah.. this is a private header. please never mind.

File: api/capi/include/tensor_filter_single.h [hunk 3]
```diff
@@ -28,8 +28,7 @@
 #include <stdint.h>
 #include <glib-object.h>
 
-#include <nnstreamer/nnstreamer_subplugin.h>
-#include <nnstreamer/nnstreamer_plugin_api_filter.h>
+#include "tensor_filter/tensor_filter_common.h"
```
Comment (myungjoo) [inline api/capi/include/tensor_filter_single.h]: uh... this doesn't look alright?
```
#include <nnstreamer/tensor_filter/tensor_filter_common.h>
```
?

**Comment by myungjoo** [comment]
LGTM other than include path designation.

File: api/capi/include/tensor_filter_single.h (hunk 3 above)
Comment (jaeyun-jung) [reply api/capi/include/tensor_filter_single.h]: Right, I just updated this PR.

**Pull request merged**